Add DNS route docs (#199)
netbirdio/docs: src/pages/how-to/routing-traffic-to-private-networks.mdx

## Concepts
### Network routes
A network route describes the network you want to connect with your NetBird peers. It has an identifier, a network range or list of domains, a routing peer, and some parameters available for managing priority and masquerading.

<Note>

>     Network routes are available for NetBird [v0.9.0](https://github.com/netbirdio/netbird/releases) or later.

Reviewer: ```suggestion
    Network routes are available in NetBird [v0.9.0](https://github.com/netbirdio/netbird/releases) or later.
```

Masquerade hides other NetBird network IPs behind the routing peer local address when accessing the target Network range. This option allows access to your private networks without configuring routes on your local routers or other devices.

If you don't enable this option, you must configure a route to your NetBird network in your external network infrastructure.

### DNS Routes

> An alternative to specifying a network range directly is to use DNS routes. Instead of adding the network directly, you can add multiple domains in a route that will be dynamically resolved on the client. The resolved IP addresses for these domains will be added as routes.

Reviewer: ```suggestion
An alternative to specifying a network range directly is to use DNS routes. Instead of adding the network directly, you can add multiple domains in a route that will be dynamically resolved on the client. The resolved IP addresses for these domains will be added as routes. For example, a network administrator can ensure that traffic to `google.com` or `api.website.com` is routed through a specific machine. So they configure DNS routes for these domains instead of specifying the IP ranges.
```

Author: I think `google.com` is not the best example as it redirects to `www.google.com` and calls a bunch of subdomains from there

Reviewer: lets use website.com :)

> The interval for resolving DNS routes is 60 seconds by default, but you can change it using the `--dns-router-interval` flag:

Reviewer 2: By default, DNS routes are resolved every 60 seconds. You can adjust this interval using the

> The interval for resolving DNS routes is 60 seconds by default, but you can change it using the `--dns-router-interval` flag:

Reviewer 2: ```suggestion
By default, DNS routes are resolved every 60 seconds. You can adjust this interval using the `--dns-router-interval` flag:
```

```bash
netbird up --dns-router-interval 30s
```

> There's also a keep routes switch that is enabled by default.

Reviewer 2: Additionally, a keep routes switch is enabled by default.

Reviewer: I suggest to highight the `Keep Routes` switch

> There's also a keep routes switch that is enabled by default.

Reviewer 2: ```suggestion
Additionally, a keep routes switch is enabled by default.
```

<p>
    <img src="/docs-static/img/how-to-guides/netbird-network-routes-dns-routes.png" alt="high-level-dia" className="imagewrapper-big"/>
</p>

> When this switch is on and a domain no longer resolves to an IP address, the corresponding route will still be kept (and any new resolved IP addresses will be added).

Reviewer 2: When the keep routes switch is on, and a domain no longer resolves to an IP address, the corresponding route will still be maintained (and any new resolved IP addresses will be added). If the switch is off, the routes will be replaced with the newly resolved IP addresses.

> When this switch is on and a domain no longer resolves to an IP address, the corresponding route will still be kept (and any new resolved IP addresses will be added).

Reviewer 2: ```suggestion
When the keep routes switch is on, and a domain no longer resolves to an IP address, the corresponding route will still be maintained (and any new resolved IP addresses will be added).
```

> If the switch is off, then the routes will be replaced with the newly resolved IP addresses instead.

Reviewer 2: If the switch is off, the routes will be replaced with the newly resolved IP addresses.

> If the switch is off, then the routes will be replaced with the newly resolved IP addresses instead.

Reviewer 2: ```suggestion
If the switch is off, the routes will be replaced with the newly resolved IP addresses.
```
